addons/hr_recruitment/views/hr_applicant_views.xml

```diff
@@ -240,7 +240,7 @@
                 <separator/>
                 <filter string="Ready for Next Stage" name="done" domain="[('kanban_state', '=', 'done')]"/>
                 <filter string="Blocked" name="blocked" domain="[('kanban_state', '=', 'blocked')]"/>
-                <filter string="Reserve" name="reserve" domain="[('job_id', '=', False)]"/>
+                <filter string="Reserve" name="reserve" domain="[('categ_ids', 'ilike', 'Reserve')]"/>
```

Comment:
I have a doubt that it will work with translations + this will include false positive like "Reservation" etc, etc.